storage/innobase/os/os0file.cc

```diff
@@ -3694,7 +3694,11 @@ os_file_create_func(
 	} else if (!srv_read_only_mode
 		   && *success
 		   && srv_unix_file_flush_method == SRV_UNIX_ALL_O_DIRECT) {
-		os_file_set_nocache(file, name, mode_str);
+		/* Do fsync() on log files when setting O_DIRECT fails.
```

Comment:
"on log and parallel doublewrite files" - see buf_parallel_dblwr_file_create, which sets O_DIRECT on the parallel doublewrite buffer, and buf_dblwr_flush_buffered_writes, which also skips flushing on SRV_UNIX_ALL_O_DIRECT. The code is correct, only the comment needs clarifying.